src/creator.js

```diff
@@ -0,0 +1,1469 @@
+				document.querySelector('.color').classList.add('show')
+				document.querySelector('#curcolor').style.background = background
+		}
+
+		return document.querySelector('#board').style.background = background
+	}
+
+	const initExistingWidget = function (title, widget, qset, version, baseUrl) {
+		qset.options.flag3D === false ? initExisting2D(title, widget, qset, version, baseUrl)
+			: initExisting3D(title, widget, qset, version, baseUrl)
+	}
+
+	function initExisting2D(title, widget, qset, version, baseUrl) {
+		mediaFileType.push('jpg', 'jpeg', 'gif', 'png')
+		_qset = qset
+
+		_setupCreator()
+		_makeDraggable()
+
+		// get asset url from Materia API (baseUrl and all)
+		const url = Materia.CreatorCore.getMediaUrl(_qset.options.image.id)
+
+		// render the image inside of the imagewrapper
+		let imageDon = document.querySelector('#image')
+		imageDon.setAttribute('src', url)
+		imageDon.setAttribute('data-imgid', _qset.options.image.id)
+
+		// load the image resource via JavaScript for rendering later
+		_img.src = url
+		let imageWrapper = document.querySelector('#imagewrapper')
+
+		_img.onload = function () {
+			imageWrapper.style.height = (_img.height * _qset.options.imageScale) + 'px'
+			return imageWrapper.style.width = (_img.width * _qset.options.imageScale) + 'px'
+		}
+
+		// set the resizable image wrapper to the size and pos from qset
+		imageWrapper.style.left = _qset.options.imageX + 'px'
+		imageWrapper.style.top = _qset.options.imageY + 'px'
+
+		// set the title from the qset
+		document.querySelector('#title').innerHTML = title
+		_title = title
+
+		// add qset terms to the list
+		// legacy support:
+		let questions = qset.items
+		if ((questions[0] != null) && questions[0].items) { questions = questions[0].items }
+
+		return Array.from(questions).map((item) =>
+			_makeTerm(item.options.endPointX, item.options.endPointY, item.questions[0].text, item.options.labelBoxX, item.options.labelBoxY, item.id))
+	}
+
+	function initExisting3D(title, widget, qset, version, baseUrl) {
+
+		_qset = qset
+
+		_setupCreator()
+		_makeDraggable()
+
+		// get asset url from Materia API (baseUrl and all)
+		const url = Materia.CreatorCore.getMediaUrl(_qset.options.image.id)
+		let realURL = 'https://'
+		let reverseSplitURL = url.split('/').reverse()
+		let httpsIndex = reverseSplitURL.findIndex(element => element === 'https:')
+
+		for (let index = httpsIndex - 2; index > -1; index--) {
+			let temp = realURL.concat(reverseSplitURL[index], '/')
+			realURL = temp
+		}
+
+		setUp3DEnvironment()
+		// PASS THE SPHERE SCALE AND UPDATE AFTER getObj() RUNS.
+		load3DAsset(realURL, _qset.options.sphereRadius)
+		_img.src = realURL
+
+		// set the title from the qset
+		document.querySelector('#title').innerHTML = title
+		_title = title
+
+		flag3D = true
+
+		// legacy support:
+		let questions = qset.items
+		if ((questions[0] != null) && questions[0].items) { questions = questions[0].items }
+
+		return setTimeout(() => {
```

Comment:
This `setTimeout()` reference looks new - but I don't think the return statement is required here, either.